app/models/labware.rb

```diff
@@ -71,14 +71,29 @@ def receptacle_proxy
 
   belongs_to :purpose, foreign_key: :plate_purpose_id, optional: true, inverse_of: :labware
 
+  # Gets the SpikedBuffer tube that is a direct parent of this labware, if it exists.
+  # The original implementation of spiked_in_buffer just supported direct parent tubes.
   has_one :spiked_in_buffer_links,
           # If we try to use the Rails 6 version, Rails 5 doesn't seem to perform the join
           -> { includes(:ancestor).references(:ancestor).where(labware: { sti_type: 'SpikedBuffer' }).direct },
           class_name: 'AssetLink',
           foreign_key: :descendant_id,
           inverse_of: :descendant
 
-  has_one :spiked_in_buffer, through: :spiked_in_buffer_links, source: :ancestor
```

Comment:
Rather than removing this relationship, I'd follow a similar parrent to the links:

```ruby
has_one :direct_spiked_in_buffer, through: :spiked_in_buffer_links, source: :ancestor
has_one :most_recent_spiked_in_buffer, through: :spiked_in_buffer_most_recent_links, source: :ancestor
```
As then the `spiked_in_buffer` method below can be simpler (See below)

Reply:
Yes OK, that's probably nicer, thanks